connectors/sources/aws.py

```diff
@@ -3,116 +3,223 @@
+
+        Yields:
+            dictionary: Document from Amazon S3.
+        """
+        bucket_list = self.configuration["buckets"] or self.get_bucket_list()
+        for bucket in bucket_list:
+            region_name = await self.get_bucket_region(bucket)
+            async with self.session.resource(
+                service_name="s3",
+                config=self.config,
+                endpoint_url=ENDPOINT_URL,
+                region_name=region_name,
+            ) as s3:
+                try:
+                    bucket_obj = await s3.Bucket(bucket)
+                    await asyncio.sleep(0)
+
+                    async for obj_summary in bucket_obj.objects.all():
+                        doc_id = md5(
+                            f"{bucket}/{obj_summary.key}".encode("utf8")
+                        ).hexdigest()
+
+                        doc = {
+                            "_id": doc_id,
+                            "filename": obj_summary.key,
+                            "size": await obj_summary.size,
+                            "bucket": bucket,
+                            "owner": (await obj_summary.owner).get("DisplayName"),
+                            "storage_class": await obj_summary.storage_class,
+                            "timestamp": (await obj_summary.last_modified).isoformat(),
+                        }
+
+                        yield doc, partial(
+                            self._get_content, doc=doc, region=region_name
+                        )
+                except Exception as exception:
+                    logger.warn(
+                        f"Something went wrong while fetching documents from {bucket}. Error: {exception}"
+                    )
 
     @classmethod
     def get_default_configuration(cls):
+        """Get the default configuration for Amazon S3.
+
+        Returns:
+            dictionary: Default configuration.
+        """
         return {
-            "bucket": {
-                "value": "ent-search-ingest-dev",
-                "label": "AWS Bucket",
-                "type": "str",
+            "buckets": {
+                "value": ["ent-search-ingest-dev"],
+                "label": "List of AWS Buckets",
+                "type": "list",
+            },
+            "read_timeout": {
+                "value": 90,
+                "label": "Read timeout",
+                "type": "int",
+            },
+            "connect_timeout": {
+                "value": 90,
+                "label": "Connection timeout",
+                "type": "int",
+            },
+            "max_attempts": {
+                "value": 5,
+                "label": "Maximum retry attempts",
+                "type": "int",
             },
-            "region": {
-                "value": "us-west-2",
-                "label": "AWS Region",
+            "connector_name": {
+                "value": "AWS Connector",
+                "label": "Friendly name for the connector",
```

Comment:
what is the purpose of `connector_name` ? I don't see it used

Reply:
@tarekziade we added this friendly name of connector here  because @danajuratoni said to add this in all the connectors.